### PR TITLE
Update to new documentation URL

### PR DESCRIPTION
**Pull request by kneth**
## Description

Fixes #

##  TODO

* [ ] Changelog entry
* [ ] Tests

### CODE REVIEW

**Review by papafe — APPROVED** [review]
Looks good! Congratulation on your first contribution 😁

**Review by nirinchev — COMMENTED** [review]
There's a bunch of API docs that also need to be updated. If you grep for `docs.mongodb.com` you can find those.

File: Realm/Realm/Exceptions/Sync/ErrorCode.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ public enum ErrorCode
     /// <c>ManualResetFallback</c> is invoked on one of the automatic client reset handlers.
     /// </summary>
     /// <seealso cref="SyncConfigurationBase.ClientResetHandler"/>
-    /// <seealso href="https://docs.mongodb.com/realm/sdk/dotnet/advanced-guides/client-reset"/>
+    /// <seealso href="https://www.mongodb.com/docs/realm/sdk/dotnet/sync/client-reset#manual-recovery/">
```
Comment (papafe) [inline Realm/Realm/Exceptions/Sync/ErrorCode.cs:67]:
```suggestion
    /// <seealso href="https://www.mongodb.com/docs/realm/sdk/dotnet/sync/client-reset#manual-recovery/"/>
```

File: Docs/index.md [hunk 2]
```diff
@@ -22,9 +22,10 @@ To see what is supported, refer to the [LINQ Support](linqsupport.md) page.
 
 Documentation
 -------------
-The documentation can be found at [https://docs.mongodb.com/realm/sdk/dotnet/](https://docs.mongodb.com/realm/sdk/dotnet/).
+The documentation can be found at [https://www.mongodb.com/docs/atlas/d
```
Comment (papafe) [inline Docs/index.md:25]: I suppose there shouldn't be a newline here

**Review by papafe — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments

File: Realm/Realm/Exceptions/Sync/ErrorCode.cs [hunk 3]
```diff
@@ -64,7 +64,7 @@ public enum ErrorCode
     /// <c>ManualResetFallback</c> is invoked on one of the automatic client reset handlers.
     /// </summary>
     /// <seealso cref="SyncConfigurationBase.ClientResetHandler"/>
-    /// <seealso href="https://docs.mongodb.com/realm/sdk/dotnet/advanced-guides/client-reset"/>
+    /// <seealso href="https://www.mongodb.com/docs/atlas/device-sdks/sdk/dotnet/sync/client-reset/#manual-recovery/">
```
Comment (papafe) [inline Realm/Realm/Exceptions/Sync/ErrorCode.cs:67]:
```suggestion
    /// <seealso href="https://www.mongodb.com/docs/atlas/device-sdks/sdk/dotnet/sync/client-reset/#manual-recovery/"/>
```

**Review by papafe — APPROVED** [review]
Apart from one comment, I think it's ready to merge

**Pull request merged**